### PR TITLE
Support mysql decimal type really

### LINKED ISSUE #205: precision issues on MariaDB 10.2.x
Server is running MariaDB 10.2.11 w/ CentOS 7.x

Table schemas:
```
CREATE TABLE `dt_decimal_1` (
  `c` decimal(38,0) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

CREATE TABLE `dt_decimal_2` (
  `c` decimal(38,19) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;

CREATE TABLE `dt_decimal_3` (
  `c` decimal(38,38) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4
```

insert data
```
INSERT INTO dt_decimal_1 VALUES (99999999999999999999999999999999999999);
INSERT INTO dt_decimal_2 VALUES (9999999999999999999.9999999999999999999);
INSERT INTO dt_decimal_3 VALUES (0.99999999999999999999999999999999999999);
```

go-canal output:
```
================================================================================
Table:          drafts.dt_decimal_1
Action:         insert
Rows:           [[1e+38]]
================================================================================
Table:          drafts.dt_decimal_2
Action:         insert
Rows:           [[1e+19]]
================================================================================
Table:          drafts.dt_decimal_3
Action:         insert
Rows:           [[1]]
```

### PR DESCRIPTION
**Pull request by arduanov**
fix #205

### CODE REVIEW

**Comment by siddontang** [comment]
Thanks @arduanov 

But this may break the compatibility, I suggest passing a config (maybe useDecimal) like another config parseTime does.

**Comment by arduanov** [comment]
Sorry for delay.
I made a changes.

File: replication/row_event_test.go [hunk 1]
```diff
@@ -133,197 +142,200 @@ func (_ *testDecodeSuite) TestDecodeDecimal(c *C) {
...
-		{[]byte{120, 99, 255, 115, 127, 255}, 7, 3, float64(-1948.140), 4, nil},
-		{[]byte{127, 255, 248, 99, 241, 127, 255}, 10, 2, float64(-1948.14), 5, nil},
-		{[]byte{127, 255, 248, 99, 255, 115, 127, 255}, 10, 3, float64(-1948.140), 6, nil},
-		{[]byte{127, 255, 255, 248, 99, 241, 118, 196}, 13, 2, float64(-1948.14), 6, nil},
-		{[]byte{118, 196, 101, 54, 0, 254, 121, 96, 127, 255}, 15, 14, float64(-9.99999999999999), 8, nil},
-		{[]byte{127, 255, 255, 248, 99, 247, 167, 196, 255, 255, 127, 255}, 20, 10, float64(-1948.1400000000), 10, nil},
-		{[]byte{127, 255, 255, 255, 255, 255, 255, 255, 255, 255, 248, 99, 255, 201, 79, 127, 255}, 30, 5, float64(-1948.14000), 15, nil},
-		{[]byte{127, 255, 255, 248, 99, 247, 167, 196, 255, 255, 255, 255, 255, 255, 127, 248}, 30, 20, float64(-1948.14000000000000000000), 14, nil},
-		{[]byte{127, 248, 99, 247, 167, 196, 255, 255, 255, 255, 255, 255, 255, 255, 255, 13, 0}, 30, 25, float64(-1948.1400000000000000000000000), 15, nil},
+		{[]byte{117, 200, 127, 255}, 4, 2, float64(-10.55), decimal.NewFromFloat(-10.55), 2, nil},
```
Comment (siddontang) [inline replication/row_event_test.go]: can we calculate the decimal directly from the expected float in the check later, so you don't add `decimal.NewFromFloat` here.

Comment (siddontang) [reply replication/row_event_test.go]: Mostly for the case, float64 and decimal have the same value.

If not, we can add another table driven test.

Comment (arduanov) [reply replication/row_event_test.go]: Maybe use strings as values and then cast to float64 and decimal?

Comment (siddontang) [reply replication/row_event_test.go]: +1

**Comment by siddontang** [comment]
Thanks @arduanov

**Pull request merged**